lib/gitsh/git_repository.rb

```diff
@@ -118,6 +123,21 @@ def abbreviated_sha
       end
     end
 
+    def modern_git_available_config_variables
+      command = git_command("config --list --name-only")
+      out, _, status = Open3.capture3(command)
+
+      if status.success?
+        out.chomp.lines.map { |line| line.chomp.to_sym }
+      end
+    end
+
+    def old_git_available_config_variables
+      git_output('config --list').lines.map do |line|
+        line.split('=').first.to_sym
```

Comment:
Sadly, that's not what `String#upto` does (e.g. `a.upto('d')` produces an enumerator over `['a', 'b', 'c', 'd']`). Your version seems more useful though.

I could use `split('=', 2).first`, that should avoid scanning the whole thing.